web-animations-1/Overview.bs

```diff
@@ -3960,9 +4111,164 @@ interface Animation : EventTarget {
     method unpauses the animation and, if the animation has already finished
     playing in the reversed direction, seeks to the start of the <a>target
     effect</a>.
+:   <dfn method for=Animation lt='persist()'>void persist()</dfn>
+::  Sets this animation's [=replace state=] to
+    [=persisted replace state|persisted=].
+:   <dfn method for=Animation lt='commitStyles()'>void commitStyles(properties)</dfn>
+::  Writes the current [=effect values=]
+    produced by this animation's [=animation effects=]
+    to their corresponding [=target elements=]' inline style
+    using the [=commit computed styles=] procedure
```

Comment:
What's the motivation for adding a commit mechanism over a way to retrieve the animation's effective output? I haven't fully thought through the implications but it seems a simpler API to simply pass a dictionary of values with the replace event.

Reply:
I believe the motivation was simply to avoid authors having to add boilerplate code. Unfortunately, passing a dictionary of values has two problems:

1. It doesn't extend easily to `GroupEffect`s. We'd need to add a `Map` or something to the event instead, to map from target elements to computed style.

2. It requires us to either _always_ flush style in order to produce the correct computed style for the current frame, or never flush style and return potentially out-of-date style (see [examples here](https://github.com/w3c/csswg-drafts/issues/3689#issuecomment-486475363)).

    At least always flushing style means we only do it once, regardless of how many animations are being removed in the current frame. Perhaps we could even spec that it only happens if there is at least one `remove` event listener, I'm not sure. Flushing style at that point in the frame might be a bit odd though -- triggering transitions before we run `requestAnimationFrame` callbacks.

    The advantage of the `commitStyles()` approach is that, even assuming we make it flush style, the author is in charge of calling it so it won't flush styles unless they use it and the side effects are more understandable. The disadvantage is that it means that if several animations are removed in the same frame, then calling `commitStyles()` several times will flush and dirty style multiple times.

Having a means to get the computed style of an `Animation` / `AnimationEffect` is nice because it might be more generally useful but it doesn't really help avoid the [boilerplate @graouts was concerned about](https://github.com/w3c/csswg-drafts/issues/3689#issuecomment-478049033).